### PR TITLE
🐛 MachinePool: always patch owned conditions, fix GetTypedPhase, doc fixes

### PR DESCRIPTION
**Pull request by sbueringer**
Signed-off-by: Stefan Büringer buringerst@vmware.com

**What this PR does / why we need it**:
While taking a closer look at MachinePools, I found a few smaller issues.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

### CODE REVIEW

File: docs/proposals/20190919-machinepool-api.md [hunk 1]
```diff
@@ -176,9 +176,6 @@ type MachinePoolSpec struct
     - Type: `MachineTemplateSpec`
     - Description: Machine Template that describes the configuration of each machine instance in a
       machine pool.
-  - **Strategy [optional]**
```
Comment (sbueringer) [inline docs/proposals/20190919-machinepool-api.md:179]: This field does not exist. Given it wasn't implemented in the last 3-4 years it probably makes sense to remove it from the proposal

File: exp/internal/controllers/machinepool_controller.go [hunk 2]
```diff
@@ -157,18 +157,16 @@ func (r *MachinePoolReconciler) Reconcile(ctx context.Context, req ctrl.Request)
 
 		// Always attempt to patch the object and status after each reconciliation.
 		// Patch ObservedGeneration only if the reconciliation completed successfully
-		patchOpts := []patch.Option{}
+		patchOpts := []patch.Option{
```
Comment (sbueringer) [inline exp/internal/controllers/machinepool_controller.go:160]: We should always patch owned conditions. Only observedGeneration should only be patched if we didn't get an error.

File: docs/proposals/20190919-machinepool-api.md (hunk 1 above)
Comment (sbueringer) [reply docs/proposals/20190919-machinepool-api.md:179]: Ah sorry, might be that we implement it now with MachinePool Machines. I'll update accordingly once I confirmed :) (still catching up..)

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/assign @Jont828 @mboersma

**Review by CecileRobertMichon — COMMENTED** [review]
/lgtm
/assign @Jont828 @mboersma

File: docs/proposals/20190919-machinepool-api.md (hunk 1 above)
Comment (sbueringer) [reply docs/proposals/20190919-machinepool-api.md:179]: Okay caught up. Sounds more like a future goal. So I think it's fine to drop it here to keep the proposal up-to-date. We can always re-add if we concretely plan do add this field.

**Review by vincepri — APPROVED** [review]
/approve

**Pull request merged**